libr/asm/p/asm_bpf_cs.c

```diff
@@ -0,0 +1,62 @@
+/* radare2 - LGPL - Copyright 2022 - terorie */
+
+#include <r_asm.h>
+#include <r_lib.h>
+#include "cs_version.h"
+
+static R_TH_LOCAL csh cd = 0;
+
+static bool the_end(void *p) {
+	if (cd) {
+		cs_close (&p);
+		p = 0;
+	}
+	return true;
+}
+
+static int disassemble(RAsm *a, RAsmOp *op, const ut8 *buf, int len) {
+	int ret, n = 0;
+
+	if (cd == 0) {
+		ret = cs_open (CS_ARCH_BPF, CS_MODE_LITTLE_ENDIAN | CS_MODE_BPF_EXTENDED, &cd);
+		if (ret != CS_ERR_OK) {
+			return -1;
+		}
+	}
+	cs_insn *insn = NULL;
+
+	op->size = 8;
+	n = cs_disasm (cd, buf, len, a->pc, 1, &insn);
+	if (n > 0 && insn->size > 0) {
+		op->size = insn->size;
+		r_strf_var (opstr, 256, "%s%s%s", insn->mnemonic, insn->op_str[0]? " ": "", insn->op_str);
+		r_asm_op_set_asm (op, opstr);
+		cs_free (insn, n);
+		return op->size;
+	}
+	r_asm_op_set_asm (op, "invalid");
+	op->size = 8;
+	cs_free (insn, n);
+	return op->size;
+}
+
+RAsmPlugin r_asm_plugin_bpf_cs = {
+	.name = "bpf",
+	.desc = "Capstone " CAPSTONE_VERSION_STRING " BPF disassembler",
+	.license = "BSD",
+	.author = "terorie",
+	.arch = "bpf",
+	.cpus = "bpf",
+	.bits = 64,
+	.endian = R_SYS_ENDIAN_LITTLE | R_SYS_ENDIAN_BIG,
```

Comment:
Should actually pass through endianness to Capstone

Reply:
Yes. But i think bpf is always big endian. Isnt?

Reply:
Capstone and LLVM support both. I'm not sure what the Linux kernel does, but Solana uses little-endian.